doc/man3/OSSL_SELF_TEST_set_callback.pod

```diff
@@ -0,0 +1,257 @@
+=pod
+
+=head1 NAME
+
+OSSL_SELF_TEST_set_callback,
+OSSL_SELF_TEST_get_callback - specify a callback for processing FIPS self tests
+
+=head1 SYNOPSIS
+
+ #include <openssl/self_test.h>
+
+ void OSSL_SELF_TEST_set_callback(OSSL_CALLBACK *cb);
+ OSSL_CALLBACK *OSSL_SELF_TEST_get_callback(void);
+
+=head1 DESCRIPTION
+
+OSSL_SELF_TEST_set_callback() sets the function B<cb> that is called during
+fips self testing. The callback function has an array of B<OSSL_PARAM> that
+contains information about the self test. Most of this information can be
+used to output information about the self tests as they run. This is useful for
+debugging if a self test is failing. The callback also allows forcing a self
+test to fail to check that it operates correctly on failure.
+
+The 'args' parameter of OSSL_CALLBACK contains the OPENSSL_CTX associated with
+the provider that is triggering the self test. This may be useful if multiple
+fips providers are present.
+
+The OSSL_PARAM names used are:
+
+=over 4
+
+=item "st-phase" (B<OSSL_PROV_PARAM_SELF_TEST_PHASE>) <UTF8 string>
+
+Each self test calls the callback 3 times with the following string values
+for the phase.
```

Comment:
...

So...  when you read a section 3 manual, you expect a reference description of the functions, and that's about it?  I find it a little unexpected to find a description of the FIPS provider in this page.  I would rather see that in a different file, and I think `doc/man7` is the appropriate place for it (nothing stops you from linking to it from the SEE ALSO in this file).  That would also allow us to add more documentation on that provider and its behaviour.

I'm unsure what name it would have, though...  `doc/man7/provider-*.pod` is kinda taken to describe the libcrypto\<-\>provider interface.  `doc/man7/FIPS-provider.pod`?  `doc/man7/OSSL_PROVIDER-FIPS.pod` (same pattern that we've used for some implementations of EVP APIs)?

Reply:
Are you saying you want all the OSSL_CALLBACK related stuff in another file?

Reply:
Uhm, not sure how to answer that question, so let me try like this... 

`OSSL_SELF_TEST_set_callback` looks like a general function that could be useful with any provider, not just our FIPS provider. Technically speaking, that is possible, and would be feasible at least with any other FIPS provider. Therefore, the general description of those functions belong here.

The description of the `OSSL_PARAM` values, however, are specific to our FIPS provider. There's nothing that I can see that obligates any other FIPS provider to pass those values. They belong in another file that describes our FIPS provider, and the programming example as well.

Reply:
While you're documenting, a description of the function added in `core_numbers.h` should be added in `doc/man7/provider-base.pod`

Reply:
I realised that there's another possible view on this; is your intention that these functions aren't as general as they appear, but are rather very specific and only useful with our FIPS provider and simply shouldn't be used with any other provider?

That *is* a possibility... But in that case, that should be made *very* clear. For the moment, nothing in this manual or in the implementation suggest such a limitation.

Reply:
docs updated